src/app/server-cluster/ServerClusterInterface.h

```diff
@@ -0,0 +1,157 @@
+                                                        AttributeValueEncoder & encoder) = 0;
+
+    /// Writes a value to an existing attribute.
+    ///
+    /// WriteAttribute MUST be done on an "existent" attribute path: only on attributes that are
+    /// returned in an `Attributes` call for this cluster.
+    ///
+    /// `request.path` is expected to have `GetClusterId` as the cluster id as well as an attribute that is
+    /// included in a `Attributes` call.
+    virtual DataModel::ActionReturnStatus WriteAttribute(const DataModel::WriteAttributeRequest & request,
+                                                         AttributeValueDecoder & decoder) = 0;
+
+    /// Retrieves the list of attributes supported by this cluster.
+    ///
+    /// Attribute list MUST contain global attributes.
+    ///
+    /// Specifically these attributes MUST always exist in the list for all clusters:
+    ///     - ClusterRevision::Id
+    ///     - FeatureMap::Id
+    ///     - AcceptedCommandList::Id
+    ///     - AttributeList::Id
+    ///     - GeneratedCommandList::Id
+    /// See SPEC 7.13 Global Elements: `Global Attributes` table
+    virtual CHIP_ERROR Attributes(const ConcreteClusterPath & path,
+                                  DataModel::ListBuilder<DataModel::AttributeEntry> & builder) = 0;
+
+    ///////////////////////////////////// Command Support /////////////////////////////////////////////////////////
+
+    /// Handles the invocation of a command.
+    ///
+    /// `handler` is used to send back the response.
+    ///    - returning `nullopt` means that return value was placed in handler directly.
+    ///      This includes cases where command handling and value return will be done asynchronously.
+    ///    - returning a value other than Success implies an error reply (error and data are mutually exclusive)
+    ///
+    /// InvokeCommand MUST be done on an "existent" attribute path: only on commands that are
+    /// returned in an `AcceptedCommand` call for this cluster.
+    ///
+    /// Return value expectations:
+    ///   - if a response has been placed into `handler` then std::nullopt MUST be returned. In particular
+    ///     note that CHIP_NO_ERROR is NOT the same as std::nullopt:
+    ///        > CHIP_NO_ERROR means handler had no status set and we expect the caller to AddStatus(success)
+    ///        > std::nullopt means that handler has added an appropriate data/status response
+    ///   - if a value is returned (not nullopt) then the handler response MUST NOT be filled. The caller
+    ///     will then issue `handler->AddStatus(request.path, <return_value>->GetStatusCode())`. This is a
+    ///     convenience to make writing Invoke calls easier.
+    virtual std::optional<DataModel::ActionReturnStatus>
+    InvokeCommand(const DataModel::InvokeRequest & request, chip::TLV::TLVReader & input_arguments, CommandHandler * handler) = 0;
```

Comment:
We should document where the TLVReader is positioned, but please do that in a followup, to avoid holding this up.